### DISCUSSION TITLE
data isolation and thread-safe when in-memory ?

### DISCUSSION THREAD

**Discussion by yellowmarlboro** [duckdb/duckdb#11970] [Q&A]
I want use duckdb in service, the service accept request from different user, i'll fetch a lot of data and append to duckdb(close the duckdb after response), and data need isolation, does duckdb had method like `duckdb_open(":memory:{requestid}")`?
And i also need thread-safe, useing(goroutine/python thread), I found some method like `cursor` in Python API.
what is the best practice ? (use `C/Python/Golang`)
thanks ~

**Comment by yellowmarlboro** [comment]
hi, anyone pls?
Also i have a question that if i was opening a duckdb(in-memory) use go-duck client, and then how can i load the db to a python duckdb-client ?
how to connect the same database in other place

**Reply by Tishj** [reply]
By using a persistent database.
in-memory connections are just that, in memory, which is bound to a process, you can't transfer that from python to go, for that you'll need to use a persistent database.

**Comment by Tishj** [comment]
We do support "named" in-memory connections, so `duckdb_open(":memory:{requestid}")` will work, and allow you to create multiple separate databases in clients like go, python, rust etc..

But it's probably more extendable to use `ATTACH` (https://duckdb.org/docs/sql/statements/attach) to attach multiple in-memory databases.
